packages/amplication-data-service-generator/src/server/resource/create-resource.ts

```diff
@@ -44,54 +46,66 @@ async function createResourceModules(
     entityName,
     entityType,
     entity,
-    dtos
+    dtos,
+    srcDirectory
   );
 
   const [serviceModule] = serviceModules;
 
-  const controllerModules = await createControllerModules(
-    appInfo,
-    resource,
-    entityName,
-    entityType,
-    serviceModule.path,
-    entity,
-    dtos
-  );
+  const controllerModules =
+    (appInfo.settings.serverSettings.generateRestApi &&
+      (await createControllerModules(
+        appInfo,
+        resource,
+        entityName,
+        entityType,
+        serviceModule.path,
+        entity,
+        dtos,
+        srcDirectory
+      ))) ||
+    [];
 
   const [controllerModule, controllerBaseModule] = controllerModules;
 
-  const resolverModules = await createResolverModules(
-    entityName,
-    entityType,
-    serviceModule.path,
-    entity,
-    dtos
-  );
+  const resolverModules =
+    (appInfo.settings.serverSettings.generateGraphQL &&
+      (await createResolverModules(
+        entityName,
+        entityType,
+        serviceModule.path,
+        entity,
+        dtos,
+        srcDirectory
+      ))) ||
+    [];
```

Comment:
I don't like the fact that we are working on two separate major changes in a single PR. This would make it much harder to track the changes in the future